### DISCUSSION TITLE
With "micro-interactions" being touted as an important part of the Windows 11 app experience, what's being done to help developers create them?

### DISCUSSION THREAD

**Discussion by mrlacey** [microsoft/WindowsAppSDK#1049] [Q&A]
Animation has traditionally been something that has been difficult to implement on Windows and developers have often avoided using them.

I expect all default controls to have these built-in, but what about custom controls? Is anything being done to make it easier for animations and micro-interactions to be created and/or applied to controls?
I've heard the claims that Lottie can be used for such effects but is this the only solution that is being proposed? While Lottie is very powerful, it can mean adding a new (another?) technology to a project and it's another thing for a developer to learn if they don't want to reuse something that already exists. It would be great if there was something to help developers building native apps with Microsoft's own technologies use what they already know.

I've deliberately not created this in the WinUI repository as the issue affects more than just WinUI apps and a (the?) solution may be created separately. Obviously, if this is only going to be addressed from a WinUI3 perspective I understand it may make sense to move this question to that repo.

**Comment by mdtauk** [comment]
There is the **AnimatedIcon** control which is being used, and I believe is triggered by the Hover and Pressed Visual States of the parent control.

I would hope more animated icons would be created for developers to use, otherwise it means creating animations in After Effects.

Also I think these Animated Icons are being used for the taskbar, but I don't think third parties can implement this.
